code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java

```diff
@@ -178,6 +196,39 @@ public void doWithLockNoResult() {
         });
     }
 
+    protected void handleServiceEvent(final ExternalEvent event, ProcessState state, ProcessInstance process) {
+        lockManager.lock(new ExternalEventLock(EXERNAL_DNS_LOCK_NAME, event.getAccountId(), event.getExternalId()),
+                new LockCallbackNoReturn() {
+                    @Override
+                    public void doWithLockNoResult() {
+                        String domainName = DataAccessor.fieldString(event, ExternalEventConstants.FIELD_DOMAIN_NAME);
+                        String serviceName = DataAccessor
+                                .fieldString(event, ExternalEventConstants.FIELD_SERVICE_NAME);
+                        String stackName = DataAccessor.fieldString(event, ExternalEventConstants.FIELD_STACK_NAME);
+                        if (domainName == null || serviceName == null || stackName == null) {
+                            return;
```

Comment:
It seems to me that it might worth logging something at info level (_maybe_ warn). Saying that you dind't have enough information and you're exiting.